Update for WWA drawing and small update for Warngen

srcarter3 opened the pull request:
WarngenDialog
- changed the button layout at the bottom to have 2 columns instead of 3, so the buttons are centered now

AbstractWWAResource.java
- Add Default value variables for the outline, shape, text, and time displays for warnings, watches, and advisories
- Add display variables for outline, shape, text, and time displays for warnings, watches, and advisories and initialize them to the default values
- Add an instance of the new DrawingPropertiesDialog so it can be disposed of properly when needed
- modified the paintInternal method
	- set the inital drawing booleans to true
	- if the record is not null and has a sig, set the drawing booleans based on their specific display booleans
	- if the entry is not null, try to draw both the shape and the outline if their respective booleans and objects are not null
	- get both the outline style and outline width from the outline capability and use that to draw the outline
	- use the text and time booleans to set the textToPrint properly and only draw what is selected
- in disposeInternal() if the drawingDialog is not null, hide it and then set it to null so it's disposed of properly when a resource is unloaded
- add setters and getters for all the display booleans and the display dialog as well

WarningsResource.java
- in initShape() add a fill (shadedshape) and outline (wireframeshape) to every entry

WatchesResource.java
- in initShape() add a outline (wireframeshape) to every entry also (already added a fill)

WouWcnWatchesResource.java
- in initShape() add a outline (wireframeshape) to every entry also (already added a fill)

WWAResourceData.java
- added a method to get the resource name (this way you can get the name without the time text appended to the end, which is useful for setting a subtitle in the drawing dialog)

DrawingPropertiesDialog.java
- created a new CAVESWTDialog that is used in the resource menu for WWA resources
- this dialog contains options for outline, fill, text, and time displays for Warnings, Watches, and Advisories

WWADrawingPropertiesAction
- created a new resource menu action that creates and assigns a new DrawingPropertiesDialog to the associated wwa resource

plugin.xml
- added the WWADrawingPropertiesAction as an "extension" at the "contextualMenu" for the option to display for WWAResource resource menus

Commit:
Update for WWA drawing and small update for Warngen

WarngenDialog
- changed the button layout at the bottom to have 2 columns instead of 3, so the buttons are centered now

AbstractWWAResource.java
- Add Default value variables for the outline, shape, text, and time displays for warnings, watches, and advisories
- Add display variables for outline, shape, text, and time displays for warnings, watches, and advisories and initialize them to the default values
- Add an instance of the new DrawingPropertiesDialog so it can be disposed of properly when needed
- modified the paintInternal method
	- set the inital drawing booleans to true
	- if the record is not null and has a sig, set the drawing booleans based on their specific display booleans
	- if the entry is not null, try to draw both the shape and the outline if their respective booleans and objects are not null
	- get both the outline style and outline width from the outline capability and use that to draw the outline
	- use the text and time booleans to set the textToPrint properly and only draw what is selected
- in disposeInternal() if the drawingDialog is not null, hide it and then set it to null so it's disposed of properly when a resource is unloaded
- add setters and getters for all the display booleans and the display dialog as well

WarningsResource.java
- in initShape() add a fill (shadedshape) and outline (wireframeshape) to every entry

WatchesResource.java
- in initShape() add a outline (wireframeshape) to every entry also (already added a fill)

WouWcnWatchesResource.java
- in initShape() add a outline (wireframeshape) to every entry also (already added a fill)

WWAResourceData.java
- added a method to get the resource name (this way you can get the name without the time text appended to the end, which is useful for setting a subtitle in the drawing dialog)

DrawingPropertiesDialog.java
- created a new CAVESWTDialog that is used in the resource menu for WWA resources
- this dialog contains options for outline, fill, text, and time displays for Warnings, Watches, and Advisories

WWADrawingPropertiesAction
- created a new resource menu action that creates and assigns a new DrawingPropertiesDialog to the associated wwa resource

plugin.xml
- added the WWADrawingPropertiesAction as an "extension" at the "contextualMenu" for the option to display for WWAResource resource menus

srcarter3 commented:
I know the comments (in the "Software History" blocks) look like they're indented differently, but when you open the file in eclipse it looks fine.

tiffanycmeyer13 commented:
Played around a little more with this, mostly everything looks good! Couple comments:
- While you can bring up the drawing properties dialog for SPS's none of the options change the SPS's. Also Sampling the text doesn't work for SPS's (this was a functionality added to other WWA by Michael James). I don't think either of these specifically needs to be addressed for this build, but something to look at in the future. Maybe just take away the drawing properties dialog for SPS? Or if it is a fairly simple fix then add it in
- Looking into the Statements - I don't think we ever plot statements in any of our bundles - something we can look at in the future if we make sure all Watches, Warnings, Advisories, and Statements are available to plot.
- I mentioned this last night, but the option for the drawing property changes/new defaults to be saved in a bundle (or procedure/display) would be ideal. I'm not exactly sure how this would work since they are all loaded in one layer, and most of the the other "capability"'s (that's what they are called in the bundle) like outline, magnification, color, are per layer. Maybe there's an option to save the desired defaults in a file locally in caveData?

Commit:
Small change for WWA functionality

SPS.xml
- enabled sampling for SPS

WWADrawingPropertiesAction
- modified isHidden() to only show for watches and warning resources, but not for CWASPSResources

srcarter3 commented:
Follow up commit was able to address:
- dialog not available for SPS
- sampling enabled for SPS
